app/models/supported_ssg.rb

```diff
@@ -87,6 +101,16 @@ def raw_supported
       # cached for the whole runtime
       @raw_supported ||= YAML.load_file(self::SUPPORTED_FILE)
     end
+
+    def build_latest_map
+      all.group_by(&:os_major_version).transform_values do |major_ssgs|
+        major_ssgs
+          .group_by(&:os_minor_version)
+          .transform_values do |minor_ssgs|
+            minor_ssgs.max_by(&:comparable_version)
+          end.freeze
+      end.freeze
```

Comment:
Could you explain the intentions behind `freeze` here? I thought it essentially makes an object immutable, which would be great, but this method will return a new object on every call. Maybe we should memoize it with `||=`?

Reply:
This is memoized and accessible within `latest_map` method. The freeze is exactly to prevent any modifications to cached/saved objects.